lib/hashme/properties.rb

```diff
@@ -68,7 +68,7 @@ def property(*args)
       def define_property_methods(property)
         # Getter
         define_method(property.name) do
-          get_attribute(property.name) || property.default
+          get_attribute(property.name)
```

Comment:
Is this correct? Might there be times when the default is not set on loading? not sure...

Reply:
I think this is correct. This was precisely the bit of code that cause the bug: if the type was a `Boolean` (or `TrueClass`) and the default was `true` then the property always returned `true` even after setting it to `false`.

I reviewed the code and the preexisting tests and I think the `set_defaults` that is called on init suffices to set the defaults properly. I couldn't think of a case where that wouldn't work. Setting the defaults on init is better because it sets them in the hash as well. It also allows to explicitly set a `nil` value to a property that has a default (or `false` like in the bug I detected)

Looking at the changes history, you [originally implemented this class](https://github.com/invopop/hashme/blame/62b20d1f33fc763275a0556bab0ceba2b2a0eb45/lib/hashme/properties.rb) only with the fallback to the default value in the getter. Then you added the `set_defaults` on init in dbc91ebaea5e4f042b0d3c1cd5f9b73e5412d493, and I wonder whether you intentionally left the getter fallback or  you simply missed to remove it.

Reply:
ahhh true story. Thanks for investigating :-) Who knows what I was thinking 9 years ago, but it does indeed look like I missed removing it 😆